contracts/Order.sol

```diff
@@ -130,42 +132,58 @@ contract Order is IERC1271, AssetRecoverer {
+     * by solvers who cannot buy tokens at the limit price when market price is higher.
      */
-    function isValidSignature(bytes32 hash_, bytes calldata) external view returns (bytes4 magicValue) {
-        if (hash_ != orderHash) revert InvalidOrderHash(orderHash, hash_);
-        if (validTo < block.timestamp) revert OrderExpired(validTo);
-
-        /// The price tolerance mechanism is crucial for ensuring that the order remains valid only within a specific price range.
-        /// This is a safeguard against market volatility and drastic price changes, which could otherwise lead to unfavorable trades.
-        /// If the price deviates beyond the tolerance level, the order is invalidated to protect against executing a trade at an undesirable rate.
-        ///
-        /// |           buyAmount                 maxToleratedAmount        currentCalculatedBuyAmount
-        /// |  --------------*-----------------------------*-----------------------------*-----------------> amount
-        /// |                 <-------- tolerance -------->
-        /// |                 <-------------------- differenceAmount ------------------->
-        ///
-        /// where:
-        ///     buyAmount - amount received from the Stonks contract, which is the minimum accepted result amount of the trade.
-        ///     tolerance - the maximum accepted deviation of the buyAmount.
-        ///     currentCalculatedBuyAmount - the currently calculated purchase amount based on real-time market conditions taken from Stonks contract.
-        ///     differenceAmount - the difference between the buyAmount and the currentCalculatedBuyAmount.
-        ///     maxToleratedAmount - the maximum tolerated deviation of the purchase amount. Represents the threshold beyond which the order is
-        ///                          considered invalid due to excessive deviation from the expected purchase amount.
+    function isValidSignature(
+        bytes32 hash_,
+        bytes calldata
+    ) external view returns (bytes4 magicValue) {
+        if (hash_ != orderHash) {
+            revert InvalidOrderHash(orderHash, hash_);
+        }
+        if (validTo < block.timestamp) {
+            revert OrderExpired(validTo);
+        }
 
         uint256 currentCalculatedBuyAmount = IStonks(stonks).estimateTradeOutput(sellAmount);
 
-        if (currentCalculatedBuyAmount <= buyAmount) return ERC1271_MAGIC_VALUE;
+        // Perfect match - accept
+        if (currentCalculatedBuyAmount == buyAmount) {
+            return ERC1271_MAGIC_VALUE;
+        }
 
-        uint256 priceToleranceInBasisPoints = IStonks(stonks).getPriceTolerance();
-        uint256 differenceAmount = currentCalculatedBuyAmount - buyAmount;
-        uint256 maxToleratedAmountDeviation = buyAmount * priceToleranceInBasisPoints / MAX_BASIS_POINTS;
+        // Reject any price improvement - makes order unfulfillable
+        if (currentCalculatedBuyAmount > buyAmount) {
```

Comment:
Possible scenario: 
1) The order placed within the price 1 ETH -> 4000 USD
2) Solver found the match in 15 minutes for this price
3) Current calculated amount decreased to the 1 ETH -> 3998 USD
4) Order is rejected

Reply:
Moreover the next condition with shortfall potentially covers this situation in a more proper way.